base/twiceprecision.jl

```diff
@@ -107,6 +107,9 @@ end
 function floatrange(a::AbstractFloat, st::AbstractFloat, len::Real, divisor::AbstractFloat)
     T = promote_type(typeof(a), typeof(st), typeof(divisor))
     m = maxintfloat(T)
```

Comment:
Hm, maybe not. At any rate, you could use a helper function to avoid having to recheck this every time you call `maxintfloat`, e.g.
```julia
_maxintfloat(T::Type) = (m = maxintfloat(T); Int === Int32 ? max(m, Float64(typemax(Int))) : m)
```
(or some better name)